common-service-core/src/main/java/org/zowe/apiml/util/CacheUtils.java

```diff
@@ -61,18 +63,15 @@ public void evictSubset(CacheManager cacheManager, String cacheName, Predicate<C
         final Cache cache = cacheManager.getCache(cacheName);
         if (cache == null) throw new IllegalArgumentException("Unknown cache " + cacheName);
         final Object nativeCache = cache.getNativeCache();
-        if (nativeCache instanceof net.sf.ehcache.Cache) {
-            final net.sf.ehcache.Cache ehCache = (net.sf.ehcache.Cache) nativeCache;
-
-            for (final Object key : ehCache.getKeys()) {
-                if (key instanceof CompositeKey) {
-                    // if entry is compositeKey and first param is different, skip it (be sure this is not to evict)
-                    final CompositeKey compositeKey = ((CompositeKey) key);
-                    if (!keyPredicate.test(compositeKey)) continue;
-                }
-                // if key is not composite key (unknown for evict) or has same serviceId, evict record
-                ehCache.remove(key);
-            }
+        if (nativeCache instanceof javax.cache.Cache) {
+            Spliterator<javax.cache.Cache.Entry<Object, Object>> spliterator = ((javax.cache.Cache<Object, Object>) nativeCache).spliterator();
```

Comment:
It is a question of course. Heap storage is based on `org.ehcache.impl.internal.concurrent.ConcurrentHashMap` and is available in the memory. The persistent memory (invalidated JWT) could be stored on the disk. In this case, reading from the disk will take the majority of the time. The amount of work depends on the amount of combination of keys (it usually depends on how many services are registered). Honestly, I don't assume any significant impact, just in point of view data are independent and we can process them parallel I have chosen this possibility, especially because caches are synchronous and it could shortcut the treatment - reduce time for blocking.

Anyway, I have no issue switching it back to sequential processing - I do not see a big difference.

Reply:
My concern was only about performance degradation due to splitting records in batches. In case you don't see or expect any impact I am ok with using this solution.